Small fixes throughout docs, re-ordering usage instructions

willium opened the pull request:
- [x] fix docs in smaller screen sizes
- [x] move usage up, rewrite
- [x] convert all double spaces to single spaces
- [x] responsive wide code blocks and wide visualizations

willium commented:
Also see: https://github.com/vega/vega-lite/issues/1226

willium commented:
Fixes #1174, #1213

kanitw commented on site/docs/using.md:
```diff
@@ -2,7 +2,7 @@
 layout: docs
 menu: docs
 title: Embedding Vega-Lite in a Web Page
```
We can use Vega-Lite in command line via vl2png, vl2svg too?   

Should we change title to Using Vega-Lite and add more sections for those?   
(And possibly add in-page TOC on the top of the page?)

kanitw replied:
or "Usage Instructions"

domoritz commented on site/docs/docs.md:
```diff
@@ -47,10 +47,3 @@ In Vega-Lite, a specification can have the following top-level properties.
 | [mark](mark.html)    | String        | The mark type.  One of `"bar"`, `"circle"`, `"square"`, `"tick"`, `"line"`, `"area"`, `"point"`, and `"text"`. |
 | [encoding](encoding.html) | Object   | A key-value mapping between encoding channels and definition of fields. |
 | [config](config.html)   | Object     | Configuration object. |
-
```
Why delete and not move up?

willium replied:
I don't think it's a good idea to have usage instructions partially in 3 places, but I can put it back (and up) if you think thats better.

kanitw replied:
I think if we're moving the menu up already, it makes sense to remove it.

domoritz replied:
So the overview page only has links to the spec docs? I have to see it to be convinced that it's a good idea.

willium commented:
I rewrote Usage Instructions, Changed Tutorials to Guides, and move it there. I know we discussed this and I believe @domoritz was against it. I can see why, but I think I like this more than before. Thoughts?

domoritz commented:
> I rewrote Usage Instructions, Changed Tutorials to Guides, and move it there. I know we discussed this and I believe @domoritz was against it. I can see why, but I think I like this more than before. Thoughts?

I can't remember whether we agreed to call the tutorials "guides"? I think I like tutorials better but we can still put a link to the usage section in the menu for tutorials. 

I'd also say that we should leave the usage.md under docs/ but just put a link to it in the tutorials menu.

* Tutorials -> shows getting started
  * Getting Started
  * Analyze data
  * Using Vega-Lite -> only a link
* Docs -> shows overview
  * Overview, needs toc
  * Using Vega-Lite
  * all the remaining docs for the spec

willium commented:
I don't really think usage really is API documentation, and I find it weird in the docs for that reason. I'm not sure I really like either, but slightly prefer Guides to Tutorials as it lends itself more to future uses of this section (integration of Vega-Lite with x, etc.) which are probably not semantically "tutorials." I suppose it could be either or.

For the TOC on Overview would it be for the entire docs? I'm not sure that's possible to generate like we do with Jekyll. How specific would be want to get

willium commented:
What about moving Usage to the highest level (next to Documentation, Examples, Tutorials in the header). To Save room we can rename Documentation to Docs, but I actually just made the header responsive so it shouldn't be too much an issue.

domoritz commented:
Usage is definitely part of docs. Docs are not only API docs imho. 

> For the TOC on Overview would it be for the entire docs? I'm not sure that's possible to generate like we do with Jekyll. How specific would be want to get

You can only generate TOC for one page so the TOC on the overview page probably has to be written by hand.

willium commented:
Okay, I'll work on the TOC. How detailed should it be?

kanitw commented:
> Usage is definitely part of docs. Docs are not only API docs imho.

Another way to see it is  the whole web is the docs.

Guide = Tutorial + Usage Guide
and we can rename docs to Syntax

kanitw commented:
This way, we don't overload the "docs" subtree with too many things.

kanitw commented on site/static/main.css:
```diff
@@ -245,6 +250,30 @@ nav a:hover {
   margin-right: 8px;
+  flex: 0 0 auto;
+  display: inline-block;
+  line-height: 3em;
```
Why is the header now taller and not centered?

kanitw commented:
> I'd also say that we should leave the usage.md under docs/ but just put a link to it in the tutorials menu.
* Tutorials -> shows getting started
  * Getting Started
  * Analyze data
  * Using Vega-Lite -> only a link
* Docs -> shows overview
  * Overview, needs toc
  * Using Vega-Lite
  * all the remaining docs for the spec


To be honest,  this is a very confusing information architecture.    I strongly suggest

1) Make docs -> syntax

2a)  Make tutorial -> guide and put usage in guide
or 2b)  Put usage as "Usage Guide" at the top-level

kanitw commented:
Thinking more about it .... Tutorial is a kind of usage guide?  so 2a) make more sense.

domoritz commented:
Docs and tutorials is more common than guides and syntax. I feel it will me more confusing even if it is more consistent. 

Let's look at some other (please help!)

* angular has top level docs with tutorial, guide (note that this is not a tutorial, it's a complete walkthrough without a running example), and more
* react has docs with quick start, guides, and reference (that's a good term, much better than "syntax"), note that guides are again not a tutorial (guides here are similar to our usage guide)
* top level documentation, then "learning rust" and references
* top level docs with "installing go", "learning go", references, articles
* top level learn with tutorial, handbook (like guides in the previous examples), samples, language spec

domoritz commented:
After looking at the examples, I propose (if we have enough space at the top level)

* Tutorials
  * Getting Started
  * Exploring Data
* Guides
  * Usage Guide
  * Faceting
  * Customizing Size
* Reference
  * Data
  * Mark
  * Encoding
  * ...

The three top level options are all docs but it makes sense to expand it.

kanitw commented:
Maybe we should have a card-sorting session.  This is indeed not an easy problem.

- Reference sounds like a more common term than syntax,  but sounds like we have to rename overview to "Specification" and move the "overview" part to guide.
- Faceting is  a weird page.  It will have also its place in "reference" after we have composition.  
- Another problem is that we wrote some language page as if they are guide.  For example, mark page is sort of a guide to all mark types.  

- I think combining tutorials and guides and make all pages except home using left side menu is another option too.  We can group the menu for guides like this:

**Tutorial** (No Link -- just category)
Getting Started
Exploring Data

**Usage Guide** (No Link -- just category)
- Embed in Web Applications
- Using on the server

**Advanced Guide**
- Faceting
- Customizing Size
- (We can add more advanced guide here)

Let's chat more tmr

kanitw commented:
Now my best proposal is we can leave it as is.  Just remove that weird "Usage Guide" jumpy link from tutorial and it kinda make sense already.  

(This way, the overview might wanna have a paragraph hinting about usage guide page though.)

kanitw commented:
Let's add a space between "Usage Instruction" and Data to suggest grouping.  

Also, we might want to make the left side menu font slightly smaller so it's smaller than the header?

willium commented:
Think this is good to merge

domoritz commented:
LGTM

domoritz commented:
One last thing. I'm sure we link to the old embed page somewhere. Please check for dead links and fix them.

kanitw commented:
LGTM too.
